cuml/forest_inference_demo.ipynb

```diff
@@ -0,0 +1,255 @@
+    "trained_model_preds, model_path = \\\n",
+    "    train_and_predict_xgboost_model(X_train, y_train,\n",
+    "                                    X_validation,\n",
+    "                                    y_validation,\n",
+    "                                    num_rounds, model_path)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### The load function of the ForestInference class accepts the following parameters:\n",
+    "        filename : str\n",
+    "           Path to saved model file in a treelite-compatible format\n",
+    "           (See https://treelite.readthedocs.io/en/latest/treelite-api.html\n",
+    "        output_class : bool\n",
+    "           If true, return a 1 or 0 depending on whether the raw prediction\n",
+    "           exceeds the threshold. If False, just return the raw prediction.\n",
+    "        threshold : float\n",
+    "           Cutoff value above which a prediction is set to 1.0\n",
+    "           Only used if the model is classification and output_class is True\n",
+    "        algo : string name of the algo from (from algo_t enum)\n",
+    "             'NAIVE' - simple inference using shared memory\n",
+    "             'TREE_REORG' - similar to naive but trees rearranged to be more\n",
+    "                              coalescing-friendly\n",
+    "             'BATCH_TREE_REORG' - similar to TREE_REORG but predicting\n",
+    "                                    multiple rows per thread block\n",
+    "        model_type : str\n",
+    "            Format of saved treelite model to load.\n",
+    "            Can be 'xgboost', 'lightgbm', or 'protobuf'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### load the saved xgboost model to return the forest in the format used as an input by the forest inference library\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fm = ForestInference.load(filename=model_path,\n",
+    "                          algo='BATCH_TREE_REORG',\n",
+    "                          output_class=True,\n",
+    "                          threshold=0.50,\n",
+    "                          model_type='xgboost')\n",
+    "# perform prediction on the model loaded from path\n",
+    "fil_preds = fm.predict(X_validation)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### check if the labels predicted with the xgboost model and the labels predicted by the ForestInference library are similar or not"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "\n",
+    "array_equal(trained_model_preds, fil_preds, tol=1e-3)"
```

Comment:
Should the arrays equal exactly? If not, I would put a simple note (in regular text size, not header) in the markdown above this cell to explain that they may not match exactly.

Reply:
Yes, they should be equal, the tolerance is required sometimes when testing for regression